src/core/account.zig

```diff
@@ -3,21 +3,25 @@ const sig = @import("../sig.zig");
 const Pubkey = sig.core.Pubkey;
 const Epoch = sig.core.Epoch;
 const AccountInFile = sig.accounts_db.accounts_file.AccountInFile;
+const ReadHandle = sig.accounts_db.buffer_pool.ReadHandle;
 
 pub const Account = struct {
     lamports: u64,
-    data: []u8,
+    data: ReadHandle,
```

Comment:
feels like this could be an enum { slice: []u8, buffer_pool: BufferPoolRead }

Reply:
this is what almost feels wrong imo - ReadHandle has a lot of methods which associate with the buffer pool whereas in this case we _only_ want to represent the data as either a slice or a buffer_pool_read and let each of those cases handle the logic themselves - separating the structs more make things easier to follow what struct contains what logic

Reply:
Hm. The point of the methods on Read Handle is to be generic, i.e. usable in cases where the data could be stored in different ways, e.g.
- deinit
- len
- read/readAll/readAllAllocate
- iterator, reader

These methods all work as expected on slices/cached reads/etc, so it's a shared API. I guess there are a few ways we could cut this?